### ISSUE TITLE
Getting warning in vs-code sveltekit project

### ISSUE THREAD

**Issue by alexbjorlig** [henriquehbr/svelte-typewriter#67]
First of all - thanks for a really awesome library - so cool feature.

When using this project I get the following warning in code: `Cannot find module 'svelte-typewriter' or its corresponding type declarations.ts(2307)`.

```
"svelte": "^3.44.0",
"@sveltejs/kit": "^1.0.0-next.232",
```

**Comment by henriquehbr** [comment]
Thanks for the kind compliments @alexbjorlig!

Can you share the link of a branch on your repo where i can reproduce this issue?

**Comment by alexbjorlig** [comment]
Added [here](https://github.com/21RISK/typewriter-repro).

1. Clone project
2. `npm install`

Typewriter actually works, but gives the following warning in index.svelte file:

![Screen Shot 2022-01-25 at 15 57 44](https://user-images.githubusercontent.com/9967422/151000850-a03497d1-57f8-4f5f-85ad-647e3d0025b2.png)

**Comment by henriquehbr** [comment]
I ran it locally, and turns out that moving `svelte-typewriter` to `dependencies` instead of `devDependencies` managed to solve the issue, can you confirm if such fix works for you as well?

I'm not exactly sure whether this is expected or not for typescript being unable to locate the definition files of `devDependencies`, anyways, that's rather odd, because usually `@types` prefixed packages are put in `devDependencies` and they work fine

I'll keep investigating that and will get back with a definitive answer as soon as possible

**Comment by alexbjorlig** [comment]
I tried moving the dependency away from dev --> but no help?
![Screen Shot 2022-01-25 at 19 12 28](https://user-images.githubusercontent.com/9967422/151034832-910b020f-8263-465b-8999-d5160617625b.png)

**Comment by henriquehbr** [comment]
My apologies, it was too precipitated of me to assume that the problems had any relation with `dependencies` or `devDependencies`, After some research, i discovered two major problems that were the culpirits of this issue:

- Turns out the directory containing the types wasn't being included on npm releases, adding `types` to the `files` field on `package.json` fixed it
- Component types were missing from the library, there were only types for the helper functions, adding them allowed Typescript to recognize `Typewriter.svelte` as containing a Svelte component

Can you please test these changes by installing `svelte-typewriter` from GitHub on the [chore/types-67](https://github.com/henriquehbr/svelte-typewriter/tree/chore/types-67) branch?

```
npm i -D henriquehbr/svelte-typewriter#chore/types-67
```

**Comment by alexbjorlig** [comment]
Hmm I can't install `npm i -D henriquehbr/svelte-typewriter#chore/types-67`, get an error related to `docker pull orhunp/git-cliff`?

**Comment by henriquehbr** [comment]
You can either install docker, or disable all `package.json` scripts (like `prepare`, which is trying to run docker) by using the `--ignore-scripts` flag, example below:

```
npm i --ignore-scripts -D henriquehbr/svelte-typewriter#chore/types-67
```

**Comment by alexbjorlig** [comment]
Hmm?

![Screen Shot 2022-01-25 at 23 41 26](https://user-images.githubusercontent.com/9967422/151071948-2f904599-2913-4318-a044-81c649beae2b.png)

**Comment by henriquehbr** [comment]
Seems like it's complaining that the project you're trying to install `svelte-typewriter` to is not a git repository, are you running that command from the `typewriter-repro` repository? i had a similar error message when executing these git hooks from inside `node_modules`

There's also a problem with the docker daemon, but i'm not exactly sure whether it's a consequence from the first error, or if it's another independent issue

**Comment by alexbjorlig** [comment]
I think the docker daemon was not running properly. Now I successfully installed after restarting the deamon - but there is now a new error: 
![Screen Shot 2022-01-25 at 23 54 40](https://user-images.githubusercontent.com/9967422/151073475-9badf5d5-b810-4d7a-961e-eaaf57dd49d0.png)

**Comment by henriquehbr** [comment]
This is weird, i'm pretty sure it has something to do about the `Typewriter.svelte` component typings not being properly recognized by Typescript, i used [sveltestrap](https://github.com/bestguy/sveltestrap) types as a basis for comparison to see if i was doing something wrong, probably there's something that i'm missing

I even tried extending the `tsconfig.json` of your project using `@tsconfig/svelte` but that didn't worked either

I'll let you know as soon as i have more updates on a fix for this

**Comment by henriquehbr** [comment]
@alexbjorlig after asking for some help for the Svelte gurus, i'm pretty sure i managed to fix it this time, can you check if this error is still showing up, please?

**Comment by alexbjorlig** [comment]
Ok - I think it's pretty close now! New error after installing with `npm i --ignore-scripts -D henriquehbr/svelte-typewriter#chore/types-67`:
![Screen Shot 2022-01-26 at 17 06 23](https://user-images.githubusercontent.com/9967422/151200463-37df8713-0e16-4608-bad6-9dc5039f3a6c.png)

**Comment by henriquehbr** [comment]
Fixed, silly mistake of mine for letting this slide in, this was happening because all of the component props were unintentionally set to be required, hence, making Typescript expect all of them to be present at once, changing them to optional by adding a `?` will hopefully address this

**Comment by alexbjorlig** [comment]
Amazing work  - no errors🔥  🎉

I don't know if you are a typescript developer yourself, but I'm telling you...
Having documentation for the properties will save you 20% issues on Github.

If you can help me get the dev env up and running, I could maybe help?

![Screen Shot 2022-01-26 at 20 59 59](https://user-images.githubusercontent.com/9967422/151237970-4de67337-48fd-4c11-b193-28c9446bc372.png)

**Comment by henriquehbr** [comment]
Sure, but before, as this issue is already solved, i'll close it and release the changes on a new version, in order to keep things focused in their respective topics, let's talk about your idea on a new issue, it sounds pretty interesting, and i'm always happy to implement new contributions :)

**Comment by alexbjorlig** [comment]
@henriquehbr sounds like a plan. Do you have a timeline for when you would publish the fix?

**Comment by henriquehbr** [comment]
Expect to see it shipped by today

**Comment by henriquehbr** [comment]
@alexbjorlig The fix for the missing component types issue is now released on npm in version [3.0.0-alpha.8](https://github.com/henriquehbr/svelte-typewriter/blob/main/CHANGELOG.md#300-alpha8---2022-01-26)

**Comment by alexbjorlig** [comment]
I don't think I have ever seen such awesome and quick open-source responsiveness - thank you soo much.  I will look into the contribution regarding the documentation
